src/main/java/org/opensearch/knn/training/TrainingDataConsumer.java

```diff
@@ -11,19 +11,30 @@
 
 package org.opensearch.knn.training;
 
-import org.apache.commons.lang.ArrayUtils;
-import org.opensearch.knn.jni.JNIService;
+import lombok.Getter;
+import lombok.Setter;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.search.SearchResponse;
 import org.opensearch.knn.index.memory.NativeMemoryAllocation;
+import org.opensearch.search.SearchHit;
 
 import java.util.List;
-import java.util.function.Consumer;
+import java.util.Map;
 
 /**
- * Transfers vectors from JVM to native memory.
+ * TrainingDataConsumer is an abstract class that defines the interface for consuming training data.
+ * It is used to process training data and add it to the training data allocation.
+ *
+ * @param <T> The type of the training data
  */
-public class TrainingDataConsumer implements Consumer<List<Float[]>> {
+public abstract class TrainingDataConsumer<T> {
```

Comment:
I feel we don't need `T` here.
I see only public abstract void processTrainingVectors(SearchResponse searchResponse, int vectorsToAdd, String fieldName); is actually used.

Reply:
updated